### ISSUE TITLE
Consider making package load optional

### ISSUE THREAD

**Issue by eddelbuettel** [klutometis/roxygen#771]
I usually call `roxygenize(roclets="rd")` (via [this script](https://github.com/eddelbuettel/littler/blob/master/inst/examples/roxy.r), usually) which worked fine for many years.  It uses `roxygenize` to translate markup from the `R/` directory into content in the `man/` directory.   This never touches `NAMESPACE` or `DESCRIPTION`.

Now, all of a sudden, the function decides to invoke `pkgload`, forcing a load/build I did not ask for, and seemingly running it with different flags (as I get compiler warnings I do not get otherwise).

I would like to suppress this package loading test.  Would you consider a PR to that effect?  Maybe skipping if `roclets` is equal to `"rd"` so that we do not need a new flag?

**Comment by eddelbuettel** [comment]
The diff you reference appears to talk about Makefiles.  

I would simply like to get back to the world where `roxygenize()` never ever attempts to compile (or load) a package.  I call it for text substitution.

**Comment by eddelbuettel** [comment]
I second the comment by @mb706, I am running into the same issue.

The bug is obviously not fixed, so I don't understand why this was closed 17 days ago.

**Comment by krlmlr** [comment]
In most cases `roxygenize()` needs to load the code, e.g. to create `\usage{}` . This has been done previously in an ad-hoc way, now _pkgload_ takes care of this.

Seems that we need better examples for the `load_code` argument to `roxygenize()` . Maybe something like this:

```r
env_legacy <- function(path) {
  env <- new.env(parent = ...)
  files <- dir(file.path(path, "R"), full.names = TRUE)
  for (file in files) source(file, local = env)
  env
}

roxygenise(..., load_code = env_legacy)
```

**Comment by eddelbuettel** [comment]
Here is another manifestation of the same bug as a fresh SO question:

https://stackoverflow.com/questions/53642735/namespace-option-created-by-rcpparmadillo-package-skeleton-causes-error/

**Comment by FlorianSchwendinger** [comment]
I see something similar (R-3.5.2 + Debian + roxygen2_6.1.1)
```r
> roxygen2::roxygenise("scs", roclets = "rd")
Loading scs
Error in getDLLRegisteredRoutines.DLLInfo(dll, addNames = FALSE) : 
  must specify DLL via a “DLLInfo” object. See getLoadedDLLs()
Calls: <Anonymous> ... assignNativeRoutines -> getDLLRegisteredRoutines.DLLInfo
```
interestingly it seems to be connected to the `useDynLib` setting in the `NAMESPACE` file.
Using `R -e 'roxygen2::roxygenise("scs", roclets = "rd")'` plus
```r
useDynLib("scs") 
export("scs", "scs_control")
```
does work. However changing the `NAMESPACE` file to
```r
useDynLib("scs", .registration = TRUE)
export("scs", "scs_control")
```
then I get the error from above.

Hope this helps to narrow down the source of the issue.

**Comment by TWilliamBell** [comment]
I believe I've encountered the same problem due to the useDynLib line not being generated in the namespace on a package using Rcpp.

**Comment by eddelbuettel** [comment]
Yes -- it is very much still an open issue and regression. It bit students in my course trying to a submit a package containing a simple Rcpp function -- some were hit by this just last evening.

**Comment by Inferrator** [comment]
Issue still present in 6.1.1
